src/format/opamTypes.mli

```diff
@@ -262,9 +273,9 @@ type 'a request = {
 
 (** user request action *)
 type user_action =
-  | Install of name_set (** The 'root' packages to be installed *)
-  | Upgrade of package_set (** The subset of packages to upgrade *)
-  | Reinstall of package_set
+  | Install of (name_set * switch_set) (** The 'root' packages to be installed *)
+  | Upgrade of (package_set * switch_set) (** The subset of packages to upgrade on each switch *)
+  | Reinstall of (package_set * switch_set)
```

Comment:
I imagines the user would ask "install package A on switch sw1 and sw2" more than "install package A on switch sw1 and package B on switch s2" . But yes, you are correct (name * switch) sets are more general, and it should not be to much of a problem to adapt the code.